Tests/src/InternalTests/SpaceEntityTests.cpp

```diff
@@ -0,0 +1,1093 @@
+/*
+ * Copyright 2024 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "Awaitable.h"
+#include "CSP/CSPFoundation.h"
+#include "CSP/Multiplayer/Components/AnimatedModelSpaceComponent.h"
+#include "CSP/Multiplayer/Components/ScriptSpaceComponent.h"
+#include "CSP/Multiplayer/MultiPlayerConnection.h"
+#include "CSP/Multiplayer/Script/EntityScript.h"
+#include "CSP/Multiplayer/SpaceEntity.h"
+#include "CSP/Systems/Script/ScriptSystem.h"
+#include "CSP/Systems/Spaces/Space.h"
+#include "CSP/Systems/SystemsManager.h"
+#include "Debug/Logging.h"
+#include "Multiplayer/NetworkEventManagerImpl.h"
+#include "Multiplayer/Script/EntityScriptBinding.h"
+#include "TestHelpers.h"
+#include "quickjspp.hpp"
+
+#include "gtest/gtest.h"
+#include <CSP/Multiplayer/Components/CustomSpaceComponent.h>
+#include <CSP/Multiplayer/Components/SplineSpaceComponent.h>
+#include <Memory/Memory.h>
+#include <PublicAPITests/SpaceSystemTestHelpers.h>
+#include <PublicAPITests/UserSystemTestHelpers.h>
+#include <atomic>
+
+using namespace csp::multiplayer;
+
+namespace
+{
+
+MultiplayerConnection* Connection;
+SpaceEntitySystem* EntitySystem;
+
+bool RequestPredicate(const csp::systems::ResultBase& Result)
+{
+	return Result.GetResultCode() != csp::systems::EResultCode::InProgress;
+}
+
+void OnUserCreated(SpaceEntity* InUser)
+{
+	EXPECT_EQ(InUser->GetComponents()->Size(), 1);
+
+	auto* AvatarComponent = InUser->GetComponent(0);
+
+	EXPECT_EQ(AvatarComponent->GetComponentType(), ComponentType::AvatarData);
+
+	std::cerr << "OnUserCreated" << std::endl;
+}
+
+} // namespace
+
+#if RUN_ALL_UNIT_TESTS || RUN_SCRIPTSYSTEM_TESTS || RUN_UPDATE_SPACE_ENTITY_GLOBAL_POSITION_TEST
```

Comment:
[Consider] It would be great to have some comments in these tests, it is hard to follow what is going on.